File: daily/121. 买卖股票的最佳时机.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(!prices.size())
            return 0;
        int diff=0;
        stack<int> stack;
        stack.push(prices[0]);
        for(int i=0;i<prices.size();i++){
            if(stack.top()>prices[i]){
                stack.pop();
                stack.push(prices[i]);
            }
            else
                diff=(prices[i]-stack.top()>diff)?prices[i]-stack.top():diff;
        }
        return diff;
    }
};

File: daily/数组/数组easy/121. 买卖股票的最佳时机.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    if(!prices.length)
        return 0;
    let diff=0,stack=[];
    stack.push(prices[0]);
    for(let i=0;i<prices.length;i++){
        if(stack[0]>prices[i]){
            stack.pop();
            stack.push(prices[i]);
        }
        else if(diff<prices[i]-stack[0])   
            diff=prices[i]-stack[0];
    }
    return diff;
};